script.js

```diff
@@ -0,0 +1,149 @@
+    'Button 1 for Popup 2',
+    'Button 2 for Popup 2',
+  );
+});
+
+const popupButton3 = document.getElementById('popupButton3');
+popupButton3.addEventListener('click', () => {
+  openPopup(
+    'TONIC',
+    ' ',
+    './images/Snapshoot 3.png',
+    './images/SnapshootPortfolio.svg',
+    'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent',
+    'Button 1 for Popup 3',
+    'Button 2 for Popup 3',
+  );
+});
+
+const popupButton4 = document.getElementById('popupButton4');
+popupButton4.addEventListener('click', () => {
+  openPopup(
+    'Multi-Post Stories',
+    ' ',
+    './images/Snapshoot Portfolio 4.png',
+    './images/SnapshootPortfolio.svg',
+    'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent',
+    'Button 1 for Popup 4',
+    'Button 2 for Popup 4',
+  );
+});
+
+closeButton.addEventListener('click', () => {
+  popupContainer.style.display = 'none';
+});
+
+const email = document.querySelector('#email');
+
+email.addEventListener('input', (event) => {
+  const correctInput = event.target.value.toLowerCase();
+  if (event.target.value !== correctInput) {
+    email.setCustomValidity('Please enter your email address using lowercase letters.');
+  } else {
+    email.setCustomValidity('');
+  }
+});
+
+// Check if there is any data in local storage
+if (localStorage.getItem('formData')) {
+  const savedData = JSON.parse(localStorage.getItem('formData'));
+  document.getElementById('name').value = savedData.name || '';
+  document.getElementById('email').value = savedData.email || '';
+  document.getElementById('message').value = savedData.message || '';
+}
+
+// Add event listeners to input fields
+const inputFields = document.querySelectorAll('input');
+inputFields.forEach((input) => {
+  input.addEventListener('change', () => {
+    const formData = { // save data in local storage
+      name: document.getElementById('name').value,
+      email: document.getElementById('email').value,
+      message: document.getElementById('message').value,
+    };
+    localStorage.setItem('formData', JSON.stringify(formData));// convert formdata object to a JSON string and save it in local store
```

Comment:
* Well done 👏 I have noticed that one of the form fields is not being porpulated upon visit below is a recording of the bug kindly fix it 

https://github.com/jovasal/jovasal.github.io/assets/52497246/ec59ccb7-b4ab-4840-bca8-231c5dce926e

Reply:
Hi dear.

Thank you so much!!

It is fixed now